packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -247,6 +247,30 @@ class WidgetTester extends WidgetController implements HitTestDispatcher, Ticker
     return TestAsyncUtils.guard<void>(() => binding.pump(duration, phase));
   }
 
+  /// Triggers a frame after `duration` amount of time, return as soon as the frame is drawn.
+  ///
+  /// This enables driving an artificially high CPU load by rendering frames in
+  /// a tight loop. It must be used with the frame policy set to
+  /// [LiveTestWidgetsFlutterBindingFramePolicy.benchmark].
+  Future<void> pumpBenchmark(Duration duration) async {
+    assert(() {
+        final TestWidgetsFlutterBinding widgetsBinding = binding;
+        return widgetsBinding is LiveTestWidgetsFlutterBinding &&
+               widgetsBinding.framePolicy == LiveTestWidgetsFlutterBindingFramePolicy.benchmark;
+    }());
+
+    dynamic caughtException;
+    void handleError(dynamic error, StackTrace stackTrace) => caughtException ??= error;
+
+    Future<void>.microtask(() { binding.handleBeginFrame(duration); }).catchError(handleError);
+    Future<void>.microtask(() { binding.handleDrawFrame(); }).catchError(handleError);
+    await idle();
```

Comment:
this uses scheduleMicrotask, whereas the original used Timer.run. Are we sure that the semantics are the same? In particular, if a microtask is scheduled by handleBeginFrame, will it run before handleDrawFrame?

We should probably add that to the test to make sure.

Reply:
Actually the semantics are not the same, thanks for pointing this out.
The solution is just to "await idle()" after "handleBeginFrame()".

I've updated the test to make sure this happens.